crates/sui-types/src/base_types.rs

```diff
@@ -199,10 +195,15 @@ impl ToAddress for PublicKeyBytes {
     }
 }
 
-impl ToAddress for PublicKey {
-    fn to_address(&self) -> SuiAddress {
-        let pk_bytes: PublicKeyBytes = self.into();
-        pk_bytes.to_address()
+impl From<&PublicKey> for SuiAddress {
```

Comment:
In rust, you generally implement `From` rather than `Into` (and then use `.into()`).
https://doc.rust-lang.org/rust-by-example/conversion/from_into.html
There's a blanket `impl<T, U> Into<U> for T
where
    U: From<T>,` in the stdlb this takes advantage of.

Reply:
Actually, I think the problem was trying to derive PublicKey and PublicKeyBytes